blacken.el

```diff
@@ -145,13 +145,45 @@ Show black output, if black exit abnormally and DISPLAY is t."
              (when display
                (pop-to-buffer errbuf))))))
 
+(defun find-nearest-file (filename &optional dir)
+  "Find files/directories like .git, pyproject.toml recursively
+   in parent directories."
+  (let* ((dirname (if dir dir (file-truename default-directory)))
+         (full-path (concat (file-name-as-directory dirname) filename))
+         (parent (file-name-directory (directory-file-name dirname))))
+    (cond
+     ((file-exists-p full-path) full-path)       ; found the file!
+     ((string= parent dirname) nil)              ; reached "/"
+     (t (find-nearest-file filename parent)))))  ; look in parent
+
+(defun read-lines-from-file (f)
+  "Reads a file and returns a list of lines in that file"
+  (split-string
+   (with-temp-buffer
+     (insert-file-contents f)
+     (buffer-substring-no-properties
+      (point-min)
+      (point-max)))
+   "\n" t))
+
+(defun line-in-file (line f)
+  "Whether a file contains a particular line."
+  (member line (read-lines-from-file f)))
+
+(defun blacken-buffer-if-project-is-blackened (&optional display)
```

Comment:
All those functions are too complicated. Working with text buffers is one of the strength of Emacs so you don't have to create a buffer, read the file, split it by lines and then look at each line just to search for something.

You can do something similar like in the [wiki elisp cookbook](https://www.emacswiki.org/emacs/ElispCookbook#toc51)
``` emacs-lisp
(with-current-buffer (find-file-noselect "~/.emacs")
    (save-excursion ;; Don't change location of point.
      (goto-char (point-min)) ;; From the beginning...
      (if (re-search-forward ".*load-path.*" nil t 1)
          (match-string-no-properties 0)
        (error "Search failed"))))
```

Reply:
I've cleaned this up -- see my note below.